src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -323,6 +345,176 @@ DatabaseShardImp::validate()
     app_.shardFamily()->reset();
 }
 
+void
+DatabaseShardImp::importNodeStore()
+{
+    std::unique_lock<std::mutex> l(m_);
+    assert(init_);
+    std::uint32_t earliestIndex;
+    std::uint32_t latestIndex;
+    {
+        auto loadLedger = [&](bool ascendSort = true) ->
+            boost::optional<std::uint32_t>
+        {
+            std::shared_ptr<Ledger> ledger;
+            std::uint32_t seq;
+            std::tie(ledger, seq, std::ignore) = loadLedgerHelper(
```

Comment:
I'm getting a crash where this call to `loadLedgerHelper` ends up trying to relock the mutex `m_`. The call stack is 

```
        rippled.exe!ripple::NodeStore::DatabaseShardImp::contains(unsigned int seq) Line 295	C++
 	rippled.exe!ripple::NodeStore::DatabaseShardImp::fetchLedger(const ripple::base_uint<256,void> & hash, unsigned int seq) Line 209	C++
 	rippled.exe!ripple::InboundLedger::init(std::unique_lock<std::recursive_mutex> & collectionLock) Line 116	C++
 	rippled.exe!ripple::InboundLedgersImp::acquire(const ripple::base_uint<256,void> & hash, unsigned int seq, ripple::InboundLedger::Reason reason) Line 94	C++
 	rippled.exe!ripple::detail::AppFamily::acquire(const ripple::base_uint<256,void> & hash, unsigned int seq) Line 100	C++
 	rippled.exe!ripple::Ledger::Ledger(const ripple::LedgerInfo & info, bool & loaded, const ripple::Config & config, ripple::Family & family, beast::Journal j) Line 265	C++
 	[External Code]	
 	rippled.exe!ripple::loadLedgerHelper(const std::basic_string<char,std::char_traits<char>,std::allocator<char> > & sqlSuffix, ripple::Application & app) Line 1176	C++
 	rippled.exe!ripple::NodeStore::DatabaseShardImp::importNodeStore::__l3::<lambda>(bool ascendSort) Line 361	C++
```

Reply:
I see the problem, thanks.